docs/requirements.txt

```diff
@@ -27,6 +27,7 @@ algoliasearch>=2.0,<3.0
 boto3>=1.24.4<2.0
 
 # build requirements
-tensorflow==2.12.0
+tensorflow==2.12.0; sys_platform != 'darwin' or platform_machine != 'arm64'
+tensorflow-macos==2.11.0; sys_platform == 'darwin' and platform_machine == 'arm64'
```

Comment:
is there a problem with `tensorflow-macos==2.12.0`? using different versions feels like a time bomb

Reply:
Yeah, 2.12.0 isn't supported on macOS w/ silicon. As far as I know, e2e_tests has conditions to use 2.11.0 for this case, so now we're just porting the same condition over to docs.

Reply:
It causes some import error of our EstimatorTrial, and needs further investgation. That can be a separate PR.